Sync/State.swift

```diff
@@ -198,6 +199,7 @@ open class Scratchpad {
         fileprivate var keyLabel: String
         var localCommands: Set<LocalCommand>
         var engineConfiguration: EngineConfiguration?
+        var enginesEnablements: [String: Bool]?
```

Comment:
It's worth commenting what these are, when they're nil or set, and what a missing engine means — true, false, or something else?